style_guides.md

```diff
@@ -0,0 +1,88 @@
+# Coding Styles and Guidelines
+**One Branch Per Feature**<br>
+When implementing different features for the project, a new branch should be created
+for each feature. When those features are finished, a pull request should be filed to
+merge that branch into the main branch. Although there are no guidelines for how to
+structure commit messages, do try to keep them as descriptive as possible.
+
+**Regarding Variable Names**<br>
+All variable names should be written in snake case while class member variables should
+consist of the "m_" prefix. However, since variables in structs are generally exposed
+outside of the class, it is fine to exclude the "m_" prefix. In snake case, all letters
+are in lower case and words are split by an underscore. See examples below:
+
+```C++
+class MyClass
+{
+public:
+	MyClass(int set_var)
+		: m_my_var(set_var) {};
```

Comment:
I think this would be clearer if the argument variable is called `my_var` instead? So we can showcase the benefit of using the `m_` prefix.